src/libsyntax_ext/assert.rs

```diff
@@ -24,6 +24,12 @@ pub fn expand_assert<'cx>(
     tts: &[TokenTree],
 ) -> Box<dyn MacResult + 'cx> {
     let mut parser = cx.new_parser_from_tts(tts);
+
+    if parser.token == token::Eof {
+        cx.span_err(sp, "requires a boolean expression");
```

Comment:
Thanks for the change!

Could you change this to something closer to

```
cx.struct_span_err(sp, "macro call requires boolean expression as argument")
    .span_label(sp, "boolean expression required")
    .emit();
```

Ideally you'd add a `Applicability::MaybeIncorrect` suggestion as well showing what the correct code would be, but that'd be overkill for this.

Reply:
Fixed.